chat-sdk-core/src/main/java/co/chatsdk/core/utils/ImageUtils.java

```diff
@@ -48,48 +54,57 @@ public static File getDiskCacheDir(Context context, String uniqueName) {
         return new File(cachePath + File.separator + uniqueName);
     }
 
-    public static File saveImageToCache (Context context, Bitmap image) {
-        File cache = getDiskCacheDir(context, ChatSDK.config().imageDirectoryName);
-        if(!cache.exists()) {
-            cache.mkdirs();
+    public static File generateImageFile(File dir, String name, String ext) {
+        if (!dir.exists()) dir.mkdirs();
+        if (name.contains(ext)) {
+            return new File(dir, name);
         }
-
-        File file = new File(cache, UUID.randomUUID() + ".png");
+        String fileName = (name += UUID.randomUUID()).replace("@", "_");
+        File file = new File(dir, fileName + ext);
         while (file.exists()) {
-            file = new File(cache, UUID.randomUUID() + ".png");
+            fileName = (name += UUID.randomUUID()).replace("@", "_");
+            file = new File(dir, fileName + ext);
         }
-
-        return saveImageToCache(context, image, file.getName());
-
+        return file;
     }
 
-    public static File saveImageToCache (Context context, Bitmap image, String name) {
-        File cache = getDiskCacheDir(context, ChatSDK.config().imageDirectoryName);
-
-        if(!cache.exists()) {
-            cache.mkdirs();
-        }
-
-        if(!name.contains(".png")) {
-            name += UUID.randomUUID() + ".png";
-        }
-
-        name = name.replace("@", "_");
-
-        File file = new File(cache, name);
+    public static File generateImageFile(Context context, String name, String ext) {
```

Comment:
Can you think of a more descriptive name for this function? It looks like it making a new empty image file.

Reply:
Maybe 'createEmptyFileInCacheDirectory'